app/controllers/admin/users_controller.rb

```diff
@@ -13,11 +13,18 @@ def new
 
   def create
     @user = User.new(user_params)
-    if @user.save
-      @user.create_with_auth0
-      redirect_to admin_user_path(@user)
-    else
+
+    User.transaction do
+      if @user.save
+        @user.create_with_auth0
+        redirect_to admin_user_path(@user)
+      else
+        render action: :new
+      end
+    rescue Auth0::Exception
+      flash[:alert] = 'There was an Auth0 error.'
```

Comment:
Perhaps we can make this a little more friendly to the admin user by suggesting what they should do about it. How about: `'There was an error adding the user to Auth0. Please try again.'`